src/core/src/core_logic/MaintenanceWindow.py

```diff
@@ -71,3 +71,27 @@ def is_package_install_time_available(self, remaining_time_in_minutes=None):
         else:
             self.composite_logger.log_warning("Time Remaining: " + str(timedelta(seconds=int(remaining_time_in_minutes * 60))) + ", Cutoff time: " + str(timedelta(minutes=cutoff_time_in_minutes)) + " [Out of time!]")
             return False
+
+    def get_percentage_maintenance_window_used(self):
+        """Calculate percentage of maintenance window used"""
+        try:
+            current_time = self.env_layer.datetime.datetime_utcnow()
+            start_time = self.env_layer.datetime.utc_to_standard_datetime(self.start_time)
+            if current_time < start_time:
+                raise Exception("Start time {} is greater than current time {}".format(str(start_time), str(current_time)))
+            dur = datetime.datetime.strptime(self.duration, "%H:%M:%S")
+            dura = timedelta(hours=dur.hour, minutes=dur.minute, seconds=dur.second)
+            total_time_in_minutes = self.env_layer.datetime.total_minutes_from_time_delta(dura)
+            if total_time_in_minutes == 0:
+                raise Exception("Total time in minutes is 0")
+            elapsed_time_in_minutes = self.env_layer.datetime.total_minutes_from_time_delta(current_time - start_time)
+            percent_maintenance_window_used = (elapsed_time_in_minutes / total_time_in_minutes) * 100
+        except Exception as error:
```

Comment:
Since you are now catching "Exception" here than a "ValueError" that was used before, we don't need to raise Exceptions on line 81 and 86. If those scenarios occur, execution will come to L89 and throw an exception here.

Also, check L89-95, it has a bunch of things we do while raising an exception, there's no action on you for that, just as a good to know thing.

Reply:
Done for line at 86.
For line at 81, we do need to throw that exception as per reason mentioned on reply on comment on line 81.